tools/functional-tester/etcd-tester/cluster.go

```diff
@@ -338,3 +343,33 @@ func (c *cluster) compactKV(rev int64) error {
 	}
 	return err
 }
+
+func (c *cluster) checkCompact(rev int64) error {
+	if rev == 0 {
+		return nil
+	}
+	for _, u := range c.GRPCURLs {
+		cli, err := clientv3.New(clientv3.Config{
+			Endpoints:   []string{u},
+			DialTimeout: 5 * time.Second,
+		})
+		if err != nil {
+			return err
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		wch := cli.Watch(ctx, "\x00", clientv3.WithFromKey(), clientv3.WithRev(rev-1))
+		wr, ok := <-wch
+		cancel()
+
+		cli.Close()
+
+		if !ok {
+			return fmt.Errorf("watch channel terminated")
+		}
+		if wr.CompactRevision != rev {
```

Comment:
It was working under my testing, but I see the hash problem. I think it works now because the hashes aren't checked until after a test case completes, so there's enough time for the compaction to go through. Should `Hash` wait until all scheduled compactions complete?